Mage/src/main/java/mage/abilities/effects/common/DrawCardSourceControllerEffect.java

```diff
@@ -57,16 +70,34 @@ public boolean apply(Game game, Ability source) {
         return false;
     }
 
-    private void createStaticText(String whoDrawCard) {
-        StringBuilder sb = new StringBuilder(whoDrawCard);
-        sb.append(whoDrawCard.isEmpty() ? "draw " : " draw ");
-        String value = amount.toString();
-        sb.append(CardUtil.numberToText(value, "a"));
-        sb.append(value.equals("1") ? " card" : " cards");
-        String message = amount.getMessage();
-        if (!message.isEmpty()) {
-            sb.append(value.equals("X") ? ", where X is " : " for each ");
-            sb.append(message);
+    private void createStaticText(boolean youDraw, DrawCardsPhrasing drawCardsPhrasing) {
+        StringBuilder sb = new StringBuilder(youDraw ? "you " : "");
+        sb.append("draw");
+        String value = " a";
+        if (amount instanceof StaticValue) {
```

Comment:
hm, I wonder if STATIC should be part of the enum, since it'll generally need its own text gen case and that way could just be part of the switch

Reply:
It's ok to use instanceOf for popular and basic classes. Just make sure it do not break cards with extended class versions.